perfkitbenchmarker/pkb.py

```diff
@@ -428,13 +437,34 @@ def RunBenchmarks(publish=True):
       vm_util.RunThreaded(
           RunBenchmark, args, max_concurrent_threads=FLAGS.parallelism)
     else:
-      for run_args, kwargs in args:
-        RunBenchmark(*run_args, **kwargs)
+      while args:
+        run_args, kwargs = args.popleft()
+        benchmark_name = run_args[0].BENCHMARK_NAME
+        try:
+          RunBenchmark(*run_args, **kwargs)
+          successful_benchmarks.append(benchmark_name)
+        except BaseException as e:
+          logging.exception(e)
```

Comment:
Nit: `logging.exception(e)` is redundant - a stacktrace will already be printed with `e` in it.
Could you add some extra information to the message, like the benchmark name, index within the collection of benchmarks running, and whether execution will continue?